### ISSUE TITLE
2.4.3 Focus order, 3.2.1 On focus,  and menus opening on focus

### ISSUE THREAD

**Issue by detlevhfischer** [w3c/wcag#4588]
The recommended pattern for a navigation menu with submenus is that the submenu opens on activation of the main menu item and the focus then moves into the submenu and on to the next main menu item.
Another pattern we find in audits is that on focusing the main menu item, the submenu automatically opens and the focus then moves into the submenu. And you have probably all seen implementations where the sdubmenu then opens further submenus that are traversed when tabbing on, leading to a situation where the time and effort to traverse the entire menu is both annoying and unpredictable. So it is clear that this pattern is undesirable (while it may be only mildly irritating if there are view submenu items and no further sub-submenus).

The specific question on which I would like to find an WG group consensus whether this behavior constitutes a clear fail of 2.4.3 Focus order or 3.2. On focus.
 
From out discussions in the WCAG 3 Input subgroup I have learned that some people are convinced that this is the case and have always treated implementations where menus open on focus as a failure of 2.4.3. Others have so far seen this as a usability issue, but not a failure of 2.4.3. Focus order or 3.2.1 On focus.

**Comment by patrickhlauke** [comment]
it would only fail 3.2.1 On Focus if context was changed automatically (generally, once you focus, it whisks you directly to the first item in the newly opened submenu).

as for 2.4.3, I would not say it's a failure (all other things being equal)

**Comment by detlevhfischer** [comment]
@patrickhlauke yes, this has been also our practice so far, since the order per se is correct, and opening a submenu on focus is not exactly unexpected…

**Comment by TestPartners** [comment]
I agree that it does not violate any WCAG success criteria. User testing would be needed in order to say which design is better with any certainty, although I would expect the first design to work better.

**Comment by mraccess77** [comment]
If escape will collapse the menu that seems to mitigate the issue some.

**Comment by detlevhfischer** [comment]
@mraccess77 
> If escape will collapse the menu that seems to mitigate the issue some.

...and it would be needed to meet 1.4.13 Content on hover and focus in cases where the popup can cover information.

**Comment by scottaohara** [comment]
agreed that an auto-opening a sub-navigation popup wouldn't be a 2.4.3 focus order failure.  

Rather the potential failure would fall more into whether the popup could be dismissed (as jon mentioned 1.4.13) _or_, a 4.1.2 issue if the automatic expansion of the content was not conveyed, and thus navigating into the popup, rather than moving to the next item in the parent list, would be unexpected due to that lack of state information.
